From 9340733377243a2f27bbc82d3799e6fc907a1963 Mon Sep 17 00:00:00 2001
From: "Pottimurthy, Sathya Narayana" <sathya.narayana.pottimurthy@intel.com>
Date: Tue, 20 Apr 2021 19:11:49 -0400
Subject: [PATCH 1/7] mcdma: DPDK MCDMA Build integration patch

This patch containd changes to integrate MCDMA
PMD to DPDK build environment

Signed-off-by: Pottimurthy, Sathya Narayana
<sathya.narayana.pottimurthy@intel.com>
Title: mcdma: DPDK MCDMA Build integration patch
---
 config/common_base   | 6 ++++++
 drivers/net/Makefile | 1 +
 mk/rte.app.mk        | 1 +
 3 files changed, 8 insertions(+)

diff --git a/config/common_base b/config/common_base
index 14000ba..47b6ab4 100644
--- a/config/common_base
+++ b/config/common_base
@@ -1124,3 +1124,9 @@ CONFIG_RTE_APP_CRYPTO_PERF=y
 # Compile the eventdev application
 #
 CONFIG_RTE_APP_EVENTDEV=y
+
+#
+##compile Intel MCDMA PMD
+##
+CONFIG_RTE_LIBRTE_MCDMA_PMD=y
+CONFIG_RTE_LIBRTE_MCDMA_DEBUG_DRIVER=n
diff --git a/drivers/net/Makefile b/drivers/net/Makefile
index 361974e..12f5b55 100644
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -62,6 +62,7 @@ DIRS-$(CONFIG_RTE_LIBRTE_THUNDERX_NICVF_PMD) += thunderx
 DIRS-$(CONFIG_RTE_LIBRTE_VDEV_NETVSC_PMD) += vdev_netvsc
 DIRS-$(CONFIG_RTE_LIBRTE_VIRTIO_PMD) += virtio
 DIRS-$(CONFIG_RTE_LIBRTE_VMXNET3_PMD) += vmxnet3
+DIRS-$(CONFIG_RTE_LIBRTE_MCDMA_PMD) += mcdma
 
 ifeq ($(CONFIG_RTE_LIBRTE_KNI),y)
 DIRS-$(CONFIG_RTE_LIBRTE_PMD_KNI) += kni
diff --git a/mk/rte.app.mk b/mk/rte.app.mk
index 77161c7..f89942d 100644
--- a/mk/rte.app.mk
+++ b/mk/rte.app.mk
@@ -98,6 +98,7 @@ _LDLIBS-$(CONFIG_RTE_LIBRTE_CMDLINE)        += -lrte_cmdline
 _LDLIBS-$(CONFIG_RTE_LIBRTE_REORDER)        += -lrte_reorder
 _LDLIBS-$(CONFIG_RTE_LIBRTE_SCHED)          += -lrte_sched
 _LDLIBS-$(CONFIG_RTE_LIBRTE_RCU)            += -lrte_rcu
+_LDLIBS-$(CONFIG_RTE_LIBRTE_MCDMA_PMD)      += -lrte_pmd_mcdma
 
 ifeq ($(CONFIG_RTE_EXEC_ENV_LINUX),y)
 _LDLIBS-$(CONFIG_RTE_LIBRTE_KNI)            += -lrte_kni
-- 
1.8.3.1

